integration/sql/iface-java/tests/java/io/openlineage/sql/OpenLineageSqlTest.java

```diff
@@ -0,0 +1,45 @@
+/*
+/* Copyright 2018-2022 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+
+package io.openlineage.sql;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+class OpenLineageSqlTest {
+    @Test
+    void basicParse() {
+        SqlMeta output = OpenLineageSql.parse(Arrays.asList("SELECT * FROM test"));
+        assertEquals(output, new SqlMeta(
+            Arrays.asList(new DbTableMeta(null, null, "test")),
+            new ArrayList<DbTableMeta>()
+        ));
+    }
+
+    @Test
+    void parseWithDialect() {
+        SqlMeta output = OpenLineageSql.parse(Arrays.asList("SELECT * FROM `random-project`.`dbt_test1`.`source_table` WHERE id = 1"), "bigquery");
+        assertEquals(output, new SqlMeta(
+            Arrays.asList(new DbTableMeta("random-project", "dbt_test1", "source_table")),
+            new ArrayList<DbTableMeta>()
+        ));
+    }
+
+    @Test
+    void parseError() {
+        boolean exceptionCaught = false;
+        try {
+            SqlMeta output = OpenLineageSql.parse(Arrays.asList("Definitely not an SQL statement"));
+        } catch (Exception e) {
+            exceptionCaught = true;
+            assertTrue(e instanceof RuntimeException);
```

Comment:
Are we able to distinguish parsing errors from errors related to library installation and platform?

Reply:
Will be done in separate PR: https://github.com/OpenLineage/OpenLineage/issues/1052